### PR TITLE
Add PointSpace fallback constructor, fix arr type

### LINKED ISSUE #1275: Missing deprecated PointSpace
## Describe the bug
We missed defining PointSpace for `AbstractPoint`s, which now results in a method error.

## To Reproduce

```julia
import ClimaCore
coord = ClimaCore.Geometry.ZPoint(Float64(1))
space = ClimaCore.Spaces.PointSpace(coord)
```

### PR DESCRIPTION
**Pull request by charleskawczynski**
Closes #1275.

### CODE REVIEW

File: src/Spaces/pointspace.jl [hunk 1]
```diff
@@ -17,17 +17,26 @@ ClimaComms.device(space::PointSpace) = ClimaComms.device(space.context)
...
 
 @deprecate PointSpace(x::Geometry.LocalGeometry) PointSpace(
-    ClimaComms.SingletonCommsContext(ClimaComms.CPUDevice()),
+    ClimaComms.CPUDevice(),
     x,
 ) false
+@deprecate PointSpace(x::Geometry.AbstractPoint) PointSpace(
+    ClimaComms.CPUDevice(),
+    x,
+) false
```
Comment (simonbyrne) [inline src/Spaces/pointspace.jl:26]: I'm still a bit reluctant to deprecate these directly...

Comment (charleskawczynski) [reply src/Spaces/pointspace.jl:26]: Sure, I think that’s fine. I’ll update

Comment (charleskawczynski) [reply src/Spaces/pointspace.jl]: Updated

**Comment by charleskawczynski** [comment]
bors try

File: src/Spaces/pointspace.jl [hunk 2]
```diff
@@ -16,18 +16,21 @@ end
...
+    context = ClimaComms.SingletonCommsContext(device)
+    return PointSpace(context, x)
+end
 
 function PointSpace(
     context::ClimaComms.AbstractCommsContext,
     local_geometry::LG,
 ) where {LG <: Geometry.LocalGeometry}
     FT = Geometry.undertype(LG)
-    # TODO: inherit array type
```
Comment (charleskawczynski) [inline src/Spaces/pointspace.jl:29]: @simonbyrne inheriting the array type ends up triggering getindex errors, should we just punt on this for now?

Comment (charleskawczynski) [reply src/Spaces/pointspace.jl:29]: I was hoping this would be an easy fix :/

Comment (charleskawczynski) [reply src/Spaces/pointspace.jl:29]: Fixed via adapt for now. I think this solution may be fine

**Comment by charleskawczynski** [comment]
bors try

**Review by simonbyrne — APPROVED** [review]

**Pull request merged**